common/src/main/java/com/linecorp/decaton/common/TaskMetadataUtil.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.decaton.common;
+
+import java.io.UncheckedIOException;
+
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.Headers;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import com.linecorp.decaton.protocol.Decaton.TaskMetadataProto;
+
+public class TaskMetadataUtil {
+    private static final String METADATA_HEADER_KEY = "dt_meta";
```

Comment:
Considering below pros/cons, I decided to choose single protobuf value.

## Decomposed primitive fields

- Pros
  * We can access each field without decoding entire metadata (not sure about the concrete use case though)
- Cons
  * Since header is just a `Map<String, byte[]>` essentially, we need to decide how to encode primitives as byte array (endianness, charset encoding, varint or fixed length, ...)
  * Also, considering when we add new metadata field in the future, we have to decide what's the "default" value
      - "default" value here means the value when a field is missing. (it happens when tasks are came from old client)

## Encoded protobuf in header

- Pros
  * Above "Cons" doesn't exist. The byte-array representation is defined by protobuf spec. Also protobuf spec defines "default" value.
- Cons
  * Above "Pros" is not possible

Reply:
> not sure about the concrete use case though

only debug cases I think.

ok, I think your point is fair. let's go with the current way then.